route_following_plugin/src/route_following_plugin.cpp

```diff
@@ -263,29 +263,36 @@ namespace route_following_plugin
     double RouteFollowingPlugin::findSpeedLimit(const lanelet::ConstLanelet& llt)
     {
         lanelet::Optional<carma_wm::TrafficRulesConstPtr> traffic_rules = wm_->getTrafficRules();
-        double target_speed;
+        double target_speed, traffic_speed =0.0, param_speed =0.0;
         double hardcoded_max=lanelet::Velocity(hardcoded_params::control_limits::MAX_LONGITUDINAL_VELOCITY_MPS * lanelet::units::MPS()).value();
 
         if (traffic_rules)
         {
-            target_speed=(*traffic_rules)->speedLimit(llt).speedLimit.value();
+            traffic_speed=(*traffic_rules)->speedLimit(llt).speedLimit.value();
             
         }
-        else
-        {
-            if(config_limit > 0 && config_limit < hardcoded_max)
-            {
-                target_speed=config_limit;
-                ROS_WARN("Failed to set the current speed limit. Valid traffic rules object could not be built. Using Configurable value");
-            }
+        else{
+            ROS_WARN(" Valid traffic rules object could not be built.");
+        }
 
-            else 
-            {
-                target_speed= hardcoded_max;
-                ROS_WARN("Failed to set the current speed limit. Valid traffic rules object could not be built. Using Hardcoded maximum");
-            }
-            
+        if(config_limit > 0.0 && config_limit < hardcoded_max)
+        {
+            param_speed = config_limit;
+            ROS_DEBUG("Using Configurable value");
+        }
+        else 
+        {
+            param_speed = hardcoded_max;
```

Comment:
I assume this is a known or intended consequence, but it does strike me as a little bit odd that if someone were to tweak the config limit here to be 0.0 that they'd end up with a limit of ~80mph if I'm understanding. That could be pretty surprising behavior for someone who thought they essentially turned off a feature by limiting it to 0.0 mph.